zkSNACKs/WalletWasabi
# QR Decoder with Emgu.CV (Alpha version)

**WIP!**

This PR improves  the first QR Reader attempt #4563
It uses the Emgu.CV QR Detector and Decoder, which is much faster and stable than the library we copied in. (#5592)

**How to test**

1. Open a Wallet.
2. Go to Send Tab
3. Fluent: There are 2 icons in the Address Panel now, click the QR Icon (second). Old GUI: There is a big QR Reader button.
4. Use your mobile to read in your QR image.
5. See that the Address label filled up with the content of the QR.
6. See that the webcam turns off after a successful reading.

**Live feed from your webcam works**.

**For Ubuntu 20.04, see @Szpoti's comment below**

**ToDo:**
- [ ] Make it work on different platforms. (Win10 and ubuntu 20.04 looks good so far)
- [x] Implement it in the Fluent Project.
- [ ] Maybe get rid of the nuget packages and import the raw code/binaries from Emgu.CV to make it more secured and less in MBs.
- [ ] If we stick with this approach then revert this #5592
- [x] Rewrite QR Code Tests (https://github.com/adamPetho/WalletWasabi/pull/2)
